src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    static final int DEFAULT_CAPACITY = 10;
+    static final float INCREASE_CAPACITY_COEFFICIENT = 1.5f;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexCheck(index);
+        checkCapacity(1);
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        checkCapacity(list.size());
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index + 1);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexCheck(index + 1);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index + 1);
+        T element = (T) elementData[index];
+        System.arraycopy(elementData, index + 1, elementData, index, size - index - 1);
+        size--;
+        return element;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = getIndexByValue(element);
+        return remove(index);
+    }
+
+    private int getIndexByValue(T value) {
+        for (int i = 0; i < size; i++) {
+            if (value == elementData[i]
+                    || (value != null && Objects.equals(value, elementData[i]))) {
+                return i;
+            }
+        }
+        throw new NoSuchElementException("No such element");
     }
```

Comment:
looks good
and also line 69
<img width="227" alt="image" src="https://user-images.githubusercontent.com/58032907/195416534-ddfe8fb2-e9f8-4b99-962a-23e1a974682d.png">

Reply:
what about my previous comment?

Reply:
ohh sorry, I've understood this after Andrew comment but forgot to delete delete Object